### PR TITLE
[all] more ldop

### PR DESCRIPTION
**Pull request by maranget**
This PR performs several enhancements related to AArch64 fetch -and-op (or AMO) instructions:
  + The unsigned minimum and maximum instructions `LDUMIN` and` LDUMAX` are added.
     As far as we know, implementation of fetch-and-op instructions is now complete.
  + `ST<OP>` is now an alias for `LD<OP>`, facilitating implementation.
  + ASL implementation of fetch-and-op instructions.
  + Bonus: the AArch64+ASL tests are now run without option  `-variant ASL`.

### CODE REVIEW

File: herd/AArch64Sem.ml [hunk 1]
```diff
@@ -1797,7 +1801,19 @@ module Make
...
       let ldop op sz rmw rs rt rn ii =
         let open AArch64 in
-        let an = rmw_to_read rmw in
+
+        let tr_input = match op with
+          |A_SMIN|A_SMAX ->
+          (* For int64 signed comparison to work on int32, int16, int8 *)
+            sxt_op sz
+          |A_ADD|A_EOR|A_SET|A_CLR|A_UMAX|A_UMIN -> M.unitT
+          and _tr_output = match op with
```
[the remark this answers is not in the document]
Comment (relokin) [reply herd/AArch64Sem.ml:1810]: also is _tr_output used anywhere? It's also the same as tr_input, isn't it?

File: lib/AArch64Base.ml [hunk 2]
```diff
@@ -1676,9 +1686,23 @@ let do_pp_instruction m =
   | I_SWPBH (bh,rmw,r1,r2,r3) ->
       sprintf "%s %s,%s,[%s]" (swpbh_memo bh rmw) (pp_wreg r1) (pp_wreg r2) (pp_xreg r3)
 (* Fecth and Op *)
+  | I_LDOP (op,v,(RMW_P|RMW_L as rmw),r1,ZR,r2) ->
+     if m.compat then
+       sprintf "%s %s,%s,[%s]"
+        (ldop_memo op rmw) (pp_vreg v r1) (pp_vreg v ZR) (pp_xreg r2)
+     else
+       sprintf "%s %s,[%s]"
+         (stop_memo op (rmw_to_w rmw)) (pp_vreg v r1) (pp_xreg r2)
```
Comment (relokin) [inline lib/AArch64Base.ml:1695]: Wouldn't this print a test with LDADD Xs,XZR,[Xn] print it as an STADD (for example in the litmus7 output), do we want that?

**Review by relokin — COMMENTED** [review]
Some minor nits otherwise LGTM!

File: internal/lib/testHerd.ml [hunk 3]
```diff
@@ -149,8 +149,13 @@ let do_check_output litmus expected  expected_failure expected_warn t =
...
   match t with
-    | _,[],[] ->
-      Printf.printf "Failed %s : Herd finished but returned no output or errors\n" litmus ; false
+    | code,[],[] ->
+       code = 0
+       ||
+         begin
+           Printf.printf "Failed %s : Herd finished but returned no output or errors\n" litmus ;
+           false
+         end
```
[the remark this answers is not in the document]
Comment (maranget) [reply internal/lib/testHerd.ml:158]: Let us follow the two of you :)

File: lib/AArch64Base.ml (hunk 2 above)
Comment (maranget) [reply lib/AArch64Base.ml:1695]: I guess we do, as documentation tells that aliases are "always the preferred disassembly".

File: herd/AArch64Sem.ml (hunk 1 above)
[the remark this answers is not in the document]
Comment (maranget) [reply herd/AArch64Sem.ml:1810]: I delete the definition of `_tr_output`. Thanks`

File: lib/AArch64Base.ml (hunk 2 above)
Comment (relokin) [reply lib/AArch64Base.ml:1695]: oh I see, sorry I missed it. Very well spotted!

**Review by relokin — APPROVED** [review]
LGTM!

**Comment by maranget** [comment]
Thanks @HadrienRenaud and @relokin. Merging on its way.

**Pull request merged**